src/service/dashboards/mod.rs

```diff
@@ -194,7 +186,11 @@ pub async fn move_dashboard(
         return Err(DashboardError::MoveMissingFolderParam);
     };
 
-    let Some(dashboard) = table::dashboards::get(org_id, from_folder, dashboard_id).await? else {
+    let Some((curr_folder, dashboard)) = table::dashboards::get(org_id, dashboard_id).await? else {
+        return Err(DashboardError::DashboardNotFound);
+    };
+
+    if curr_folder.folder_id != from_folder {
         return Err(DashboardError::DashboardNotFound);
     };
```

Comment:
We do this extra check here now because before `table::dashboards::get(org_id, from_folder, dashboard_id).await?` returning `Some` would tell us that "the dashboard exists AND it exists in the source folder".

But now `table::dashboards::get(org_id, dashboard_id).await?` returning `Some` only tells us that "the dashboard exists in SOME folder". Therefore we must also check that the current folder is, in fact, the desired source folder. If it's not then it's possible that the folder was already moved - so we shouldn't try to move it again.